clang/lib/AST/ASTImporter.cpp

```diff
@@ -9896,9 +9903,13 @@ Expected<FileID> ASTImporter::Import(FileID FromID, bool IsBuiltin) {
         // FIXME: The filename may be a virtual name that does probably not
         // point to a valid file and we get no Entry here. In this case try with
         // the memory buffer below.
-        if (Entry)
+        if (Entry) {
+          if (isBufferSizeOverflow(ToSM, Entry->getSize()))
+            return llvm::make_error<ASTImportError>(
+                ASTImportError::UnsupportedConstruct);
```

Comment:
It is better to use `Unknown` as error kind here (it is used already in this function), or add a new type of error to `ASTImportError::ErrorKind`.

Reply:
`ASTImportError::Unknown` would crash CTU module too, `ASTImportError::UnsupportedConstruct` might be deliberately chosen so the analysis could continue in this case.

Reply:
This error is clearly not an unsupported AST construct. For this case a new error type should be added like `ASTImportError::SourceLocationOverflow` or `ResourceOverflow`. Code in `CrossTranslationUnitContext::importDefinitionImpl` should be updated to count this new error.
Alternatively the `llvm_unreachable` from `UnknownError` can be removed. But `UnknownError` should be used if something happens that is not supposed to happen, even if now it is used like "other error".

Reply:
> This error is clearly not an unsupported AST construct. For this case a new error type should be added like ASTImportError::SourceLocationOverflow or ResourceOverflow

Agreed